src/Utilities/Compiler/Extensions/CompilationExtensions.cs

```diff
@@ -59,5 +61,15 @@ internal static bool TargetsDotNetFramework(this Compilation compilation)
 
             return false;
         }
+
+        /// <summary>
+        /// Gets a value indicating whether the project of the compilation is a Web SDK project,
+        /// based on MSBuild property <see cref="MSBuildPropertyOptionNames.UsingMicrosoftNETSdkWeb"/>.
+        /// </summary>
+        internal static bool IsWebProject(this Compilation compilation, AnalyzerOptions options, CancellationToken cancellationToken)
+        {
+            var usingMicrosoftNETSdkWeb = options.GetMSBuildPropertyValue(MSBuildPropertyOptionNames.UsingMicrosoftNETSdkWeb, compilation, cancellationToken);
```

Comment:
[cosmetic] the variable name reads to me as if it's a boolean while it's a string, maybe `usingMicrosoftNETSdkWebPropertyValue` is clearer (not sure because it's really long...).